#3649 Update CI templates

uilianries opened the pull request:
Hi this PR brings some updates related to issue #3649 :

- CONAN_REFERENCE is no longer required
- Add GCC 8 on default CI template list
- Add Clang 5, 6 on default CI template list
- Update unit test to avoid CONAN_REFERENCE

closes #3649 

Changelog: Fix

- [X] Refer to the issue that supports this Pull Request.
- [X] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [X] I've read the [Contributing guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [X] I've followed the PEP8 style guides for Python code.
- [ ] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one.

uilianries commented:
My changes didn't affect Macos jobs :thinking: 

infrastructure error?

danimtb commented:
yes, that is what I meant

lasote commented on conans/client/cmd/new_ci.py:
```diff
@@ -128,7 +126,6 @@
 gitlab = """
 variables:
     CONAN_USERNAME: "{user}"
-    CONAN_REFERENCE: "{name}/{version}"
```
I'm not sure if we should remove this. It is not necessary if the recipe declares the name and version but necessary otherwise.

uilianries replied:
correct, I almost forgot about that. taocpp does not use name and version in recipes, so conan reference is mandatory. I'll revert this line.

lasote commented:
Should we add clang 7?

Croydon commented:
We should first solve the discussion around the new Clang 7 version schema I guess.

uilianries commented:
Did you mean #3643 ?

Croydon commented:
> Did you mean #3643 ?

Yes

Croydon commented:
I can't see any change related to

> CONAN_REFERENCE is no longer required

:thinking:

uilianries commented:
> I can't see any change related to
> 
> > CONAN_REFERENCE is no longer required
> 
> thinking

I removed at first moment, but I reverted after Lasote request:

https://github.com/conan-io/conan/pull/3651/commits/806fce7b23d0445aaef952b0f4df2d5190f3d63a

uilianries commented:
@danimtb 

> Please update Appveyor CI template to use Python 3 instead of python 2

Done!

Croydon commented:
Could you also please update the container names to `conanio/<compiler><compiler-version>`? It seems still to use the old names

uilianries commented:
> Could you also please update the container names to `conanio/<compiler><compiler-version>`? It seems still to use the old names

sure! thanks for reviewing.

uilianries commented:
@Croydon Done!

uilianries commented:
I forgot to update the tests :man_facepalming:
